src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -0,0 +1,289 @@
+        {"exception_code", std::make_shared<DataTypeEnum8>(getZooKeeperErrorEnumsAndValues())},
+    };
+}
+
+
+static bool extractPathImpl(const IAST & elem, String & res, const Context & context)
+{
+    const auto * function = elem.as<ASTFunction>();
+    if (!function)
+        return false;
+
+    if (function->name == "and")
+    {
+        for (const auto & child : function->arguments->children)
+            if (extractPathImpl(*child, res, context))
+                return true;
+        return false;
+    }
+
+    if (function->name == "equals")
+    {
+        const auto & args = function->arguments->as<ASTExpressionList &>();
+        ASTPtr value;
+
+        if (args.children.size() != 2)
+            return false;
+
+        const ASTIdentifier * ident;
+        if ((ident = args.children.at(0)->as<ASTIdentifier>()))
+            value = args.children.at(1);
+        else if ((ident = args.children.at(1)->as<ASTIdentifier>()))
+            value = args.children.at(0);
+        else
+            return false;
+
+        if (ident->name() != "cluster")
+            return false;
+
+        auto evaluated = evaluateConstantExpressionAsLiteral(value, context);
+        const auto * literal = evaluated->as<ASTLiteral>();
+        if (!literal)
+            return false;
+
+        if (literal->value.getType() != Field::Types::String)
+            return false;
+
+        res = literal->value.safeGet<String>();
+        return true;
+    }
+    return false;
+}
+
+
+/** Retrieve from the query a condition of the form `path = 'path'`, from conjunctions in the WHERE clause.
+  */
+static String extractPath(const ASTPtr & query, const Context & context)
+{
+    const auto & select = query->as<ASTSelectQuery &>();
+    if (!select.where())
+        return "";
+    String res;
+    return extractPathImpl(*select.where(), res, context) ? res : "";
+}
+
+void StorageSystemDDLWorkerQueue::fillData(MutableColumns & res_columns, const Context & context, const SelectQueryInfo & query_info) const
+{
+    String cluster_name = extractPath(query_info.query, context);
+    if (cluster_name.empty())
+        throw Exception(
+            "SELECT from system.ddl_worker_queue table must contain condition like cluster = 'cluster' in WHERE clause.",
+            ErrorCodes::BAD_ARGUMENTS);
+
+    zkutil::ZooKeeperPtr zookeeper = context.getZooKeeper();
+    Coordination::Error zk_exception_code = Coordination::Error::ZOK;
+    String ddl_zookeeper_path = config.getString("distributed_ddl.path", "/clickhouse/task_queue/ddl/");
+    String ddl_query_path;
+
+    // this is equivalent to query zookeeper at the `ddl_zookeeper_path`
+    /* [zk: localhost:2181(CONNECTED) 51] ls /clickhouse/task_queue/ddl
+       [query-0000000000, query-0000000001, query-0000000002, query-0000000003, query-0000000004]
+    */
+
+    // if there is error querying the root path of the ddl directory throw exception and stop
```

Comment:
But now it returns 0 rows instead of throwing exception if we cannot get list of entries in queue

Reply:
Yes, this comment is not accurate has to be removed. Initially I had thought about throwing an exception if we aren't able to query node information anywhere in the code flow. But if we cannot retrieve the list of entries for a single node then we won't be able to fill the `distributed_ddl_queue` for all other nodes as well.
I will remove the comment and instead leave the rows empty on failure.